homeassistant/components/madvr/sensor.py

```diff
@@ -0,0 +1,239 @@
+        translation_key="outgoing_colorimetry",
+    ),
+    MadvrSensorEntityDescription(
+        key="outgoing_black_levels",
+        name="Outgoing Black Levels",
+        value_fn=lambda coordinator: coordinator.data.get("outgoing_black_levels"),
+        translation_key="outgoing_black_levels",
+    ),
+    MadvrSensorEntityDescription(
+        key="aspect_res",
+        name="Aspect Ratio Resolution",
+        value_fn=lambda coordinator: coordinator.data.get("aspect_res"),
+        translation_key="aspect_res",
+    ),
+    MadvrSensorEntityDescription(
+        key="aspect_dec",
+        name="Aspect Ratio Decimal",
+        value_fn=lambda coordinator: coordinator.data.get("aspect_dec"),
+        translation_key="aspect_dec",
+    ),
+    MadvrSensorEntityDescription(
+        key="aspect_int",
+        name="Aspect Ratio Integer",
+        value_fn=lambda coordinator: coordinator.data.get("aspect_int"),
+        translation_key="aspect_int",
+    ),
+    MadvrSensorEntityDescription(
+        key="aspect_name",
+        name="Aspect Ratio Name",
+        value_fn=lambda coordinator: coordinator.data.get("aspect_name"),
+        translation_key="aspect_name",
+    ),
+    MadvrSensorEntityDescription(
+        key="masking_res",
+        name="Masking Resolution",
+        value_fn=lambda coordinator: coordinator.data.get("masking_res"),
+        translation_key="masking_res",
+    ),
+    MadvrSensorEntityDescription(
+        key="masking_dec",
+        name="Masking Decimal",
+        value_fn=lambda coordinator: coordinator.data.get("masking_dec"),
+        translation_key="masking_dec",
+    ),
+    MadvrSensorEntityDescription(
+        key="masking_int",
+        name="Masking Integer",
+        value_fn=lambda coordinator: coordinator.data.get("masking_int"),
+        translation_key="masking_int",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: MadVRConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor entities."""
+    coordinator = entry.runtime_data
+    async_add_entities(MadvrSensor(coordinator, description) for description in SENSORS)
+
+
+class MadvrSensor(MadVREntity, SensorEntity):
+    """Base class for madVR sensors."""
+
+    def __init__(
+        self,
+        coordinator: MadVRCoordinator,
+        description: MadvrSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.entity_description: MadvrSensorEntityDescription = description
+        self._attr_unique_id = f"{coordinator.mac}_{description.key}"
+
+    @property
+    def native_value(self) -> float | str | None:
+        """Return the state of the sensor."""
+        return self.entity_description.value_fn(self.coordinator)
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.native_value is not None
```

Comment:
if native_value is `None` it is rendered as `unknown` and that is a good thing imo.